### PR TITLE
Fixing loss function code in tutorial

### PR DESCRIPTION
**Pull request by mrkumar83**
The function softmax_cross_entropy_loss is no longer available in gluon.loss. Making the necessary update to the tutorial code.

Also, record() in one part of the tutorial is not preceded by autograd resulting in an undefined error, adding the reference to autograd.

### CODE REVIEW

File: docs/tutorials/gluon/gluon.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ To compute loss and backprop for one iteration, we do:
 label = mx.nd.arange(10)  # dummy label
 with autograd.record():
     output = net(data)
-    loss = gluon.loss.softmax_cross_entropy_loss(output, label)
+    loss = gluon.loss.SoftmaxCrossEntropyLoss()(output, label)
```
Comment (szha) [inline docs/tutorials/gluon/gluon.md]: loss = gluon.loss.SoftmaxCrossEntropyLoss(output, label)

File: docs/tutorials/gluon/gluon.md [hunk 2]
```diff
@@ -127,9 +127,9 @@ this is a commonly used functionality, gluon provide a `Trainer` class for it:
 ```python
 trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': 0.01})
 
-with record():
+with autograd.record():
     output = net(data)
-    loss = gluon.loss.softmax_cross_entropy_loss(output, label)
+    loss = gluon.loss.SoftmaxCrossEntropyLoss()(output, label)
```
Comment (szha) [inline docs/tutorials/gluon/gluon.md]: same
